cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/PartitionedFileSetSource.java

```diff
@@ -0,0 +1,127 @@
+    configurer.addProperty(new Property(PARTITIONING, "Partitioning to use for the Partitioned FileSet. " +
+      "Must be provided if you want the pipeline to create the fileset. ex: " +
+      "{\"league\" : \"string\", \"season\" : \"int\"}", false));
+    configurer.addProperty(new Property(TextOutputFormat.SEPERATOR, "The output format separator must be provided " +
+      "if you want the pipeline to create the fileset", false));
+
+  }
+
+  @Override
+  public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineConfigurer) {
+    // if the partitioning is provided then we should try to create the fileset here
+    if (!Strings.isNullOrEmpty(stageConfig.getProperties().get(PARTITIONING))) {
+      String pfsName = stageConfig.getProperties().get(FILESET_NAME);
+      String separator = stageConfig.getProperties().get(TextOutputFormat.SEPERATOR);
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(pfsName), "PartitionedFileSet name must be given.");
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(pfsName), "A separator for " +
+        TextOutputFormat.SEPERATOR + "must be given");
+      pipelineConfigurer.createDataset(pfsName, PartitionedFileSet.class.getName(),
+                                       PartitionedFileSetProperties.builder()
+        .setPartitioning(ETLUtils.createPartitioning(stageConfig.getProperties().get(PARTITIONING)))
+        .setInputFormat(TextInputFormat.class)
+        .setOutputFormat(TextOutputFormat.class)
+        .setOutputProperty(TextOutputFormat.SEPERATOR, separator)
+        .build());
+    }
+  }
+
+  @Override
+  public void prepareJob(BatchSourceContext context) {
+    Map<String, String> inputArgs = Maps.newHashMap();
+    PartitionedFileSetArguments.setInputPartitionFilter(inputArgs,
+                                                        createPartitionFilter(
+                                                          context.getRuntimeArguments().get(FILTER)));
+    PartitionedFileSet input = context.getDataset(context.getRuntimeArguments().get(FILESET_NAME), inputArgs);
+    context.setInput(context.getRuntimeArguments().get(FILESET_NAME), input);
+  }
+
+  /**
+   * creates a {@link PartitionFilter} from a given string in JSON format
+   *
+   * @param filterString partition filter JSON string
+   * @return {@link PartitionFilter}
+   */
+  private PartitionFilter createPartitionFilter(String filterString) {
+    PartitionFilter.Builder builder = PartitionFilter.builder();
+    Type stringStringMap = new TypeToken<Map<String, String>>() {
+    }.getType();
+    Map<String, String> filters = new Gson().fromJson(filterString, stringStringMap);
+    for (Map.Entry<String, String> entrySet : filters.entrySet()) {
+      String key = entrySet.getKey();
+      if (key.endsWith(VALUE)) {
+        builder.addValueCondition(key.split("\\.")[0], entrySet.getValue());
```

Comment:
if it is valid for a partition key to contain a '.', this will break.  It is better to find the last to take a substring from the beginning to the end - length of VALUE